src/nvim/tui/tui.c

```diff
@@ -935,7 +934,8 @@ static void tui_set_mode(UI *ui, ModeShape mode)
     }
   }
 
-  switch (shape) {
+  int shape;
```

Comment:
maybe default to something, though I guess @bfredl's PR prevents the need for that.

Reply:
It's an enum. The compiler will catch if a unhandled value gets added to the enum. (I tested)